fix the PullRequest object so it can parse

adnelson opened the pull request:
I'm not positive this covers every variant of a pull request event, but prior to this change the `FromJSON` instance failed to decode the [example json](https://developer.github.com/v3/activity/events/types/#pullrequestevent) (which is where the JSON I included here comes from) in the API docs, and now it does.

adnelson commented on src/GitHub/Data/PullRequests.hs:
```diff
@@ -160,6 +159,9 @@ data PullRequestEventType
     | PullRequestUnassigned
     | PullRequestLabeled
     | PullRequestUnlabeled
+    | PullRequestReviewRequested
```
added these because the API docs list them.

Not sure if it makes sense to make a catch-all `OtherPullRequestEventType Text` which might make the library a little more future-proof.

adnelson commented:
Not sure why the PR builder is unable to find `fixtures/pull-request.json`; it's in the same directory as other fixtures and it worked locally

phadej commented on src/GitHub/Data/PullRequests.hs:
```diff
@@ -191,7 +193,7 @@ instance FromJSON SimplePullRequest where
         <*> o .: "html_url"
         <*> o .: "updated_at"
         <*> o .:? "body"
-        <*> o .: "assignees"
```
I'm 100% sure that API returns a list, and GitHub docs are wrong. You can have multiple assignees for PR/Issue.

phadej replied:
Check: https://api.github.com/repos/phadej/github/pulls/285, there are both keys "assignee" and "assignees"

adnelson commented:
What do you mean by using the real world response as a fixture?

phadej commented:
@adnelson e.g. the current version of result for that the PR in your fixture https://api.github.com/repos/baxterthehacker/public-repo/pulls/1 has `assignees` list.

Real-world: don't copy old fixtures from somewhere, grab new ones from the API.

adnelson commented:
@phadej I think the problem was that I was confusing the `PullRequest` and `PullRequestEvent` types. I'll add fixtures for each case and test them separately

adnelson commented:
Alright, so I added three more fixtures, one that I had of the pull request event, and the two that you had given for pull requests, making four test cases in total. I added some logic which allows either the `assignee` or `assignees` keys, or both, to be present in a pull request blob.

phadej commented on src/GitHub/Data/PullRequests.hs:
```diff
@@ -74,12 +77,10 @@ data PullRequest = PullRequest
     , pullRequestCommits        :: !Count
     , pullRequestMerged         :: !Bool
     , pullRequestMergeable      :: !(Maybe Bool)
-    , pullRequestMergeableState :: !MergeableState
```
why this is removed? (and review comments?)

adnelson replied:
ah sorry, that was due to my confusion between pull request / pull request event. I'll put it back in

phadej commented on src/GitHub/Data/PullRequests.hs:
```diff
@@ -180,8 +184,22 @@ instance Binary PullRequestReference
 
+-- | Helper function, reads either the "assignee" OR "assigneed" OR
+-- both from a JSON object.
+getAssignees :: Object -> Parser (Vector SimpleUser)
```
I'd trust there's always `assignees`, if there and endpoint which doesn't return plural `assignees`?

adnelson replied:
[this example](https://developer.github.com/v3/pulls/#list-pull-requests) has an `assignee` key but no `assignees` keys.

phadej replied:
But it's current variant https://api.github.com/repos/octocat/Hello-World/pulls **does** have `assignees`. *GitHub documentation is outdated, because examples are manually written*.

adnelson replied:
Is there a downside to including both? If we include both then it will work either way. You yourself earlier posted an example which contained both keys. I'm fine removing it but this seems like the way to accommodate all configurations which are likely to appear.

adnelson replied:
@phadej if it's a dealbreaker for getting this merged then I can remove the `assignee` part

adnelson commented:
@phadej any thoughts on this?

adnelson commented:
@phadej just checking in here...

andreasabel commented:
@adnelson : New maintainer here.  
Are you willing to fix the conflicts so that this PR can be merged (assuming it is still serving its purpose)?

andreasabel reviewed (changes requested):
Please resolve conflicts.
